File: 10-1.rb
#!/usr/bin/env ruby

ratings = [0] + File.read('10.input').lines.map(&:to_i).sort
ratings += [ratings[-1] + 3]

threes = 0
ones = 0
row = 1

while row < ratings.size
  case ratings[row] - ratings[row - 1]
  when 1
    ones += 1
  when 3
    threes += 1
  end
  row += 1
end

print "#{ones * threes}\n"

File: 10-1.py
#!/usr/bin/env python

ratings = [0] + sorted([int(row) for row in open('10.input').readlines()])
ratings += [ratings[-1] + 3]

threes = 0
ones = 0
row = 1

while row < len(ratings):
    if ratings[row] - ratings[row - 1] == 1:
        ones += 1
    elif ratings[row] - ratings[row - 1] == 3:
        threes += 1

    row += 1

print(f"{ones * threes}")
